tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -958,7 +975,565 @@ int main(int argc, char **argv)
+
+                /* Reset handshake values */
+                conn->handshake.client_hello_received = 0;
+                conn->client_hello.parsed = 0;
+                POSIX_CHECKED_MEMSET((uint8_t*) conn->extension_requests_received, 0, sizeof(s2n_extension_bitfield));
+            }
+
+            /* Server sends HelloRetryRequest */
+            EXPECT_SUCCESS(s2n_server_hello_retry_send(server_conn));
+
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&client_conn->handshake.io));
+            EXPECT_SUCCESS(s2n_stuffer_copy(&server_conn->handshake.io, &client_conn->handshake.io,
+                                            s2n_stuffer_data_available(&server_conn->handshake.io)));
+            client_conn->handshake.message_number = HELLO_RETRY_MSG_NO;
+
+            /* Client receives HelloRetryRequest */
+            EXPECT_FAILURE_WITH_ERRNO(s2n_server_hello_recv(client_conn),
+                                      S2N_ERR_BAD_MESSAGE);
+        }
+    }
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc8446#4.1.4
+     *# Upon receipt of a HelloRetryRequest, the client MUST check the
+     *# legacy_version, legacy_session_id_echo, cipher_suite, and
+     *# legacy_compression_method as specified in Section 4.1.3 and then
+     *# process the extensions, starting with determining the version using
+     *# "supported_versions".
+     *
+     * The client MUST Process extensions, starting with supported_versions
+     *
+     *= https://tools.ietf.org/rfc/rfc8446#4.1.4
+     *= type=test
+     *# Otherwise, the client MUST process all extensions in the
+     *# HelloRetryRequest and send a second updated ClientHello.
+     **/
+     {
+         DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key,
+                       s2n_cert_chain_and_key_ptr_free);
+         EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                                                        S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+         DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                       s2n_config_ptr_free);
+         EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+         EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));
+         EXPECT_SUCCESS(s2n_config_disable_x509_verification(config));
+
+         /* Ensure the supported_versions extension has been processed */
+         {
+             DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                           s2n_connection_ptr_free);
+             EXPECT_NOT_NULL(server_conn);
+             EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+             DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                           s2n_connection_ptr_free);
+             EXPECT_NOT_NULL(server_conn);
```

Comment:
```suggestion
             EXPECT_NOT_NULL(client_conn);
```
Actually I'm going to stop commenting this since this was probably a copy and paste error, and now you know to fix the ones below.